File: NetFramework/TreeView/VB/BoundModeWithBindingList/Products/Store.vb
Imports System.ComponentModel

Public Class Store
    Public Property ID As Guid
    Public Property Name As String
    Public Property ProductsGroups As BindingList(Of ProductsGroup)
    Public ReadOnly Property CountOfProducts As Int32
        Get
            Dim count = 0
            For Each pg As ProductsGroup In ProductsGroups
                count += pg.CountOfProducts
            Next
            Return count
            End Get
    End Property

    Public Sub New(name As String)

        Me.ID = Guid.NewGuid()
        Me.Name = name
        Me.ProductsGroups = New BindingList(Of ProductsGroup)
    End Sub

    Public Overrides Function ToString() As String
        Return Name
    End Function

End Class

File: NetFramework/TreeView/CS/BoundModeWithBindingList/Products/Store.cs
using System;
using System.ComponentModel;

namespace SamplesData
{
    public class Store
    {
        public Guid ID { get; set; }
        public string Name { get; set; }
        public BindingList<ProductsGroup> ProductsGroups { get; set; }
        public int CountOfProducts
        {
            get
            {
                int count = 0;
                foreach (var pg in ProductsGroups)
                    count += pg.CountOfProducts;
                return count;
            }
        }

        public Store(string name)
        {
            ID = Guid.NewGuid();
            Name = name;
            ProductsGroups = new BindingList<ProductsGroup>();
        }

        /// <summary>
        /// Returns string representation.
        /// </summary>
        public override string ToString()
        {
            return Name;
        }        
    }
}
